core/chain_indexer.go

```diff
@@ -44,6 +44,10 @@ type ChainIndexerBackend interface {
 
 	// Commit finalizes the section metadata and stores it into the database.
 	Commit() error
+
+	// If a functions blocks for a longer period, it should immediately return
+	// with an error when Cancel is called.
+	Cancel()
```

Comment:
Does this actually terminate the indexer backend? I'm a bit worried about the semantics here, because `Cancel` is meaningful in the scope of a single request, but what happens with a background process? I could understand a `Close` that terminates everything, but the notion of Canceling some internal op seems strange as the internal state will end up corrupted.

TL;DR: Don't we really want `Close() error` here?

Reply:
Shutdown is still handled by the indexer, this is just a signal for the backend to not block on external events anymore (like ODR retrievals). I agree about semantics though, I changed the name and the description to better reflect this behavior.